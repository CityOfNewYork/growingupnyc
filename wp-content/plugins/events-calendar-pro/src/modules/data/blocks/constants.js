//
// ─── BLOCK TYPES ────────────────────────────────────────────────────────────────
//

export const RECURRING = 'recurring';
export const EXCEPTION = 'exception';

export const BLOCK_TYPES = [ RECURRING, EXCEPTION ];

//
// ─── STATE KEYS ─────────────────────────────────────────────────────────────────
//

export const KEY_TYPE = 'type';
export const KEY_ALL_DAY = 'all_day';
export const KEY_MULTI_DAY = 'multi_day';
export const KEY_MULTI_DAY_SPAN = 'multi_day_span';
export const KEY_START_TIME = 'start_time';
export const KEY_END_TIME = 'end_time';
export const KEY_START_TIME_INPUT = '_start_time_input';
export const KEY_END_TIME_INPUT = '_end_time_input';
export const KEY_START_DATE = 'start_date';
export const KEY_START_DATE_INPUT = '_start_date_input';
export const KEY_START_DATE_OBJ = '_start_date_obj';
export const KEY_END_DATE = 'end_date';
export const KEY_END_DATE_INPUT = '_end_date_input';
export const KEY_END_DATE_OBJ = '_end_date_obj';
export const KEY_LIMIT = 'limit';
export const KEY_LIMIT_DATE_INPUT = '_limit_date_input';
export const KEY_LIMIT_DATE_OBJ = '_limit_date_obj';
export const KEY_LIMIT_TYPE = 'limit_type';
export const KEY_BETWEEN = 'between';
export const KEY_DAYS = 'days';
export const KEY_WEEK = 'week';
export const KEY_DAY = 'day';
export const KEY_MONTH = 'month';
export const KEY_TIMEZONE = 'timezone';
