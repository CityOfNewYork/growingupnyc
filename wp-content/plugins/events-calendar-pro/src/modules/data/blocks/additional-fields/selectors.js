/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import identity from 'lodash/identity';
import includes from 'lodash/includes';
import uniq from 'lodash/uniq';

/**
 * Internal dependencies
 */
import { constants } from '@moderntribe/common/data/plugins';

export const getPlugin = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ];
export const getBlocks = createSelector(
	[ getPlugin ],
	( plugin ) => plugin.blocks,
);

export const getAdditionalFields = createSelector(
	[ getBlocks ],
	( blocks ) => blocks.additionalFields,
);

export const getFieldName = ( state, props ) => props.name;

export const getIds = createSelector(
	[ getAdditionalFields ],
	( fields ) => fields.allIds,
);

export const getFieldAsObjects = createSelector(
	[ getAdditionalFields ],
	( fields ) => fields.byId,
);

export const getFieldBlock = createSelector(
	[ getFieldAsObjects, getFieldName ],
	( fields, name ) => fields[ name ] || {},
);

export const getFieldDividerList = createSelector(
	[ getFieldBlock ],
	( field ) => field.dividerList,
);

export const getFieldDividerEnd = createSelector(
	[ getFieldBlock ],
	( field ) => field.dividerEnd,
);

export const getFieldOutput = createSelector(
	[ getFieldBlock ],
	( field ) => field.output,
);

export const getFieldMetaKey = createSelector(
	[ getFieldBlock ],
	( field ) => field.metaKey,
);

export const getFieldType = createSelector(
	[ getFieldBlock ],
	( field ) => field.type,
);

export const getFieldLabel = createSelector(
	[ getFieldBlock ],
	( field ) => field.label || '',
);

export const getFieldValue = createSelector(
	[ getFieldBlock ],
	( field ) => field.value,
);

export const getTextFieldValue = createSelector(
	[ getFieldValue ],
	( value ) => value || '',
);

export const getTextAreaOutput = createSelector(
	[ getFieldOutput ],
	( output = '' ) => {
		return output.split( '\n' ).filter( identity );
	},
);

export const getFieldIsPristine = createSelector(
	[ getFieldBlock ],
	( field ) => field.isPristine,
);

export const getFieldOptions = createSelector(
	[ getFieldBlock ],
	( field ) => field.options || [],
);

export const getFieldOptionsWithLabels = createSelector(
	[ getFieldOptions ],
	( options ) => {
		return options.map( ( option ) => ( { value: option, label: option } ) );
	},
);

export const getFieldDropdownValue = createSelector(
	[ getFieldBlock ],
	( field ) => ( { value: field.value, label: field.value } ),
);

export const getFieldDropdownOutput = createSelector(
	[ getFieldBlock ],
	( field ) => field.output,
);

export const getFieldCheckboxValue = createSelector(
	[ getTextFieldValue ],
	( value ) => uniq( value.split( '|' ) ),
);

export const getFieldCheckboxOptions = createSelector(
	[ getFieldCheckboxValue, getFieldOptionsWithLabels ],
	( values, optionsWithLabels ) => {
		return optionsWithLabels.map( ( option ) => ( {
			...option,
			isChecked: includes( values, option.value ),
		} ) );
	},
);
