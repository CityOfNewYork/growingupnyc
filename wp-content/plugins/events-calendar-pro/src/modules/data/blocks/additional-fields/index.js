/**
 * Internal dependencies
 */
import reducer from './reducers';
import * as types from './types';
import * as actions from './actions';
import * as selectors from './selectors';
import sagas from './sagas';

export default reducer;
export { types, actions, selectors, sagas };
