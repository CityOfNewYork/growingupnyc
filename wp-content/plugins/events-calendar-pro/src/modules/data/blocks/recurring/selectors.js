/* eslint-disable max-len */

/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import { find } from 'lodash';

/**
 * Internal dependencies
 */
import {
	RECURRENCE_TYPE_RULES_OPTIONS,
	SERIES_ENDS_OPTIONS,
} from './options';
import { constants } from '@moderntribe/common/data/plugins';
import * as selectors from '@moderntribe/events-pro/data/shared/selectors';

export const getRules = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ].blocks.recurring;
export const getRulesCount = createSelector( getRules, ( rules ) => rules.length );
export const hasRules = createSelector( getRulesCount, ( count ) => !! count );
export const getIndex = ( _, props ) => props.index;

export const getRule = createSelector(
	[ getRules, getIndex ],
	( rules, index ) => rules[ index ],
);

export const getType = createSelector( getRule, selectors.getType );
export const getAllDay = createSelector( getRule, selectors.getAllDay );
export const getMultiDay = createSelector( getRule, selectors.getMultiDay );
export const getMultiDaySpan = createSelector( getRule, selectors.getMultiDaySpan );
export const getStartDate = createSelector( getRule, selectors.getStartDate );
export const getStartDateObj = createSelector( getRule, selectors.getStartDateObj );
export const getStartDateInput = createSelector( getRule, selectors.getStartDateInput );
export const getStartTime = createSelector( getRule, selectors.getStartTime );
export const getStartTimeNoSeconds = createSelector( getRule, selectors.getStartTimeNoSeconds );
export const getStartTimeInput = createSelector( getRule, selectors.getStartTimeInput );
export const getEndDate = createSelector( getRule, selectors.getEndDate );
export const getEndDateObj = createSelector( getRule, selectors.getEndDateObj );
export const getEndDateInput = createSelector( getRule, selectors.getEndDateInput );
export const getEndTime = createSelector( getRule, selectors.getEndTime );
export const getEndTimeNoSeconds = createSelector( getRule, selectors.getEndTimeNoSeconds );
export const getEndTimeInput = createSelector( getRule, selectors.getEndTimeInput );
export const getBetween = createSelector( getRule, selectors.getBetween );
export const getLimitType = createSelector( getRule, selectors.getLimitType );
export const getLimit = createSelector( getRule, selectors.getLimit );
export const getLimitDateObj = createSelector( getRule, selectors.getLimitDateObj );
export const getLimitDateInput = createSelector( getRule, selectors.getLimitDateInput );
export const getDays = createSelector( getRule, selectors.getDays );
export const getDay = createSelector( getRule, selectors.getDay );
export const getMonth = createSelector( getRule, selectors.getMonth );
export const getWeek = createSelector( getRule, selectors.getWeek );
export const getTimezone = createSelector( getRule, selectors.getTimezone );

export const getTypeOption = createSelector(
	[ getType ],
	( type ) => find( RECURRENCE_TYPE_RULES_OPTIONS, ( option ) => option.value === type ),
);

export const getLimitTypeOption = createSelector(
	[ getLimitType ],
	( limitType ) => find( SERIES_ENDS_OPTIONS, ( option ) => option.value === limitType ),
);
