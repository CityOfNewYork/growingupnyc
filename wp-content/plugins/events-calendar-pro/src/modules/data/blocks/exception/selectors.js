/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import { find } from 'lodash';

/**
 * Internal dependencies
 */
import {
	EXCEPTION_OCCURRENCE_OPTIONS,
} from './options';
import { SERIES_ENDS_OPTIONS } from '@moderntribe/events-pro/data/blocks/recurring/options';
import { constants } from '@moderntribe/common/data/plugins';
import * as selectors from '@moderntribe/events-pro/data/shared/selectors';

export const getExceptions = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ].blocks.exception;
export const getRules = getExceptions;
export const getExceptionsCount = createSelector( getExceptions, ( exceptions ) => exceptions.length );
export const hasExceptions = createSelector( getExceptionsCount, ( count ) => !! count );
export const getIndex = ( _, props ) => props.index;

export const getRule = createSelector(
	[ getExceptions, getIndex ],
	( exceptions, index ) => exceptions[ index ],
);

export const getType = createSelector( getRule, selectors.getType );
export const getAllDay = createSelector( getRule, selectors.getAllDay );
export const getMultiDay = createSelector( getRule, selectors.getMultiDay );
export const getMultiDaySpan = createSelector( getRule, selectors.getMultiDaySpan );
export const getStartDate = createSelector( getRule, selectors.getStartDate );
export const getStartDateObj = createSelector( getRule, selectors.getStartDateObj );
export const getStartDateInput = createSelector( getRule, selectors.getStartDateInput );
export const getStartTime = createSelector( getRule, selectors.getStartTime );
export const getStartTimeNoSeconds = createSelector( getRule, selectors.getStartTimeNoSeconds );
export const getStartTimeInput = createSelector( getRule, selectors.getStartTimeInput );
export const getEndDate = createSelector( getRule, selectors.getEndDate );
export const getEndDateObj = createSelector( getRule, selectors.getEndDateObj );
export const getEndDateInput = createSelector( getRule, selectors.getEndDateInput );
export const getEndTime = createSelector( getRule, selectors.getEndTime );
export const getEndTimeNoSeconds = createSelector( getRule, selectors.getEndTimeNoSeconds );
export const getEndTimeInput = createSelector( getRule, selectors.getEndTimeInput );
export const getBetween = createSelector( getRule, selectors.getBetween );
export const getLimitType = createSelector( getRule, selectors.getLimitType );
export const getLimit = createSelector( getRule, selectors.getLimit );
export const getLimitDateObj = createSelector( getRule, selectors.getLimitDateObj );
export const getLimitDateInput = createSelector( getRule, selectors.getLimitDateInput );
export const getDays = createSelector( getRule, selectors.getDays );
export const getDay = createSelector( getRule, selectors.getDay );
export const getMonth = createSelector( getRule, selectors.getMonth );
export const getWeek = createSelector( getRule, selectors.getWeek );
export const getTimezone = createSelector( getRule, selectors.getTimezone );

export const getTypeOption = createSelector(
	[ getType ],
	( type ) => find( EXCEPTION_OCCURRENCE_OPTIONS, ( option ) => option.value === type ),
);

export const getLimitTypeOption = createSelector(
	[ getLimitType ],
	( limitType ) => find( SERIES_ENDS_OPTIONS, ( option ) => option.value === limitType ),
);
