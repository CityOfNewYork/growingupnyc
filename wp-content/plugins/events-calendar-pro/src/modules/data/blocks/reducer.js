/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import recurring from '@moderntribe/events-pro/data/blocks/recurring/reducer';
import exception from '@moderntribe/events-pro/data/blocks/exception/reducer';
import additionalFields from '@moderntribe/events-pro/data/blocks/additional-fields';

export default combineReducers( {
	recurring,
	exception,
	additionalFields,
} );
