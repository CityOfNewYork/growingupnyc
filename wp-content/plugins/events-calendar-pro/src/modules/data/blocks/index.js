/**
 * Internal dependencies
 */
import * as constants from './constants';
import * as recurring from './recurring';
import reducer from './reducer';

export default reducer;
export { constants };
export { recurring };
