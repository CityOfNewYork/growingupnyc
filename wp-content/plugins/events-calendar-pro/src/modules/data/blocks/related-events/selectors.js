/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import { constants } from '@moderntribe/common/data/plugins';

export const getRelatedEventsBlock = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ].blocks.relatedEvents;

export const getTitle = createSelector(
	[ getRelatedEventsBlock ],
	( relatedEvents ) => relatedEvents.title,
);

export const getDisplayImages = createSelector(
	[ getRelatedEventsBlock ],
	( relatedEvents ) => relatedEvents.displayImages,
);
