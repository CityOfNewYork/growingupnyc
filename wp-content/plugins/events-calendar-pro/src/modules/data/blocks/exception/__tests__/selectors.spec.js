/**
 * Internal Dependencies
 */
import { constants as pluginsConstants } from '@moderntribe/common/data/plugins';
import { constants as blocksConstants } from '@moderntribe/events-pro/data/blocks';
import * as selectors from '../selectors';

describe( 'Exception Selectors', () => {
	let exception, state, ownProps;
	beforeEach( () => {
		exception = {
			[ blocksConstants.KEY_TYPE ]: 'single',
			[ blocksConstants.KEY_ALL_DAY ]: false,
			[ blocksConstants.KEY_MULTI_DAY ]: true,
			[ blocksConstants.KEY_START_DATE ]: '2018-09-21',
			[ blocksConstants.KEY_START_DATE_INPUT ]: 'September 21, 2018',
			[ blocksConstants.KEY_START_DATE_OBJ ]: new Date( 'September 21, 2018' ),
			[ blocksConstants.KEY_START_TIME ]: '14:53',
			[ blocksConstants.KEY_START_TIME_INPUT ]: '14:53',
			[ blocksConstants.KEY_END_DATE ]: '2019-05-14',
			[ blocksConstants.KEY_END_DATE_INPUT ]: 'May 14, 2019',
			[ blocksConstants.KEY_END_DATE_OBJ ]: new Date( 'May 14, 2019' ),
			[ blocksConstants.KEY_END_TIME ]: '09:43',
			[ blocksConstants.KEY_END_TIME_INPUT ]: '09:43',
			[ blocksConstants.KEY_BETWEEN ]: 1,
			[ blocksConstants.KEY_LIMIT_TYPE ]: 'count',
			[ blocksConstants.KEY_LIMIT ]: 7,
			[ blocksConstants.KEY_LIMIT_DATE_INPUT ]: 'May 14, 2019',
			[ blocksConstants.KEY_LIMIT_DATE_OBJ ]: new Date( 'May 14, 2019' ),
			[ blocksConstants.KEY_DAYS ]: [],
			[ blocksConstants.KEY_WEEK ]: 'first',
			[ blocksConstants.KEY_DAY ]: 1,
			[ blocksConstants.KEY_MONTH ]: [],
			[ blocksConstants.KEY_TIMEZONE ]: 'UTC',
			[ blocksConstants.KEY_MULTI_DAY_SPAN ]: 'next_day',
		};
		state = {
			[ pluginsConstants.EVENTS_PRO_PLUGIN ]: {
				blocks: {
					exception: [ exception ],
				},
			},
		};
		ownProps = {
			index: 0,
		};
	} );

	const keys = Object.keys( selectors );

	keys.forEach( ( key ) => {
		test( key, () => {
			expect( selectors[ key ]( state, ownProps ) ).toMatchSnapshot();
		} );
	} );
} );
