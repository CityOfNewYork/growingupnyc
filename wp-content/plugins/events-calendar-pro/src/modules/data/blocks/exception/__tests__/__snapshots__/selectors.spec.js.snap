// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exception Selectors getAllDay 1`] = `false`;

exports[`Exception Selectors getBetween 1`] = `1`;

exports[`Exception Selectors getDay 1`] = `1`;

exports[`Exception Selectors getDays 1`] = `Array []`;

exports[`Exception Selectors getEndDate 1`] = `"2019-05-14"`;

exports[`Exception Selectors getEndDateInput 1`] = `"May 14, 2019"`;

exports[`Exception Selectors getEndDateObj 1`] = `2019-05-14T00:00:00.000Z`;

exports[`Exception Selectors getEndTime 1`] = `"09:43"`;

exports[`Exception Selectors getEndTimeNoSeconds 1`] = `"09"`;

exports[`Exception Selectors getExceptions 1`] = `
Array [
  Object {
    "_end_date_input": "May 14, 2019",
    "_end_date_obj": 2019-05-14T00:00:00.000Z,
    "_limit_date_input": "May 14, 2019",
    "_limit_date_obj": 2019-05-14T00:00:00.000Z,
    "_start_date_input": "September 21, 2018",
    "_start_date_obj": 2018-09-21T00:00:00.000Z,
    "all_day": false,
    "between": 1,
    "day": 1,
    "days": Array [],
    "end_date": "2019-05-14",
    "end_time": "09:43",
    "limit": 7,
    "limit_type": "count",
    "month": Array [],
    "multi_day": true,
    "multi_day_span": "next_day",
    "start_date": "2018-09-21",
    "start_time": "14:53",
    "timezone": "UTC",
    "type": "single",
    "week": "first",
  },
]
`;

exports[`Exception Selectors getExceptionsCount 1`] = `1`;

exports[`Exception Selectors getIndex 1`] = `0`;

exports[`Exception Selectors getLimit 1`] = `7`;

exports[`Exception Selectors getLimitDateInput 1`] = `"May 14, 2019"`;

exports[`Exception Selectors getLimitDateObj 1`] = `2019-05-14T00:00:00.000Z`;

exports[`Exception Selectors getLimitType 1`] = `"count"`;

exports[`Exception Selectors getLimitTypeOption 1`] = `
Object {
  "label": "After",
  "value": "count",
}
`;

exports[`Exception Selectors getMonth 1`] = `Array []`;

exports[`Exception Selectors getMultiDay 1`] = `true`;

exports[`Exception Selectors getMultiDaySpan 1`] = `"next_day"`;

exports[`Exception Selectors getRule 1`] = `
Object {
  "_end_date_input": "May 14, 2019",
  "_end_date_obj": 2019-05-14T00:00:00.000Z,
  "_limit_date_input": "May 14, 2019",
  "_limit_date_obj": 2019-05-14T00:00:00.000Z,
  "_start_date_input": "September 21, 2018",
  "_start_date_obj": 2018-09-21T00:00:00.000Z,
  "all_day": false,
  "between": 1,
  "day": 1,
  "days": Array [],
  "end_date": "2019-05-14",
  "end_time": "09:43",
  "limit": 7,
  "limit_type": "count",
  "month": Array [],
  "multi_day": true,
  "multi_day_span": "next_day",
  "start_date": "2018-09-21",
  "start_time": "14:53",
  "timezone": "UTC",
  "type": "single",
  "week": "first",
}
`;

exports[`Exception Selectors getRules 1`] = `
Array [
  Object {
    "_end_date_input": "May 14, 2019",
    "_end_date_obj": 2019-05-14T00:00:00.000Z,
    "_limit_date_input": "May 14, 2019",
    "_limit_date_obj": 2019-05-14T00:00:00.000Z,
    "_start_date_input": "September 21, 2018",
    "_start_date_obj": 2018-09-21T00:00:00.000Z,
    "all_day": false,
    "between": 1,
    "day": 1,
    "days": Array [],
    "end_date": "2019-05-14",
    "end_time": "09:43",
    "limit": 7,
    "limit_type": "count",
    "month": Array [],
    "multi_day": true,
    "multi_day_span": "next_day",
    "start_date": "2018-09-21",
    "start_time": "14:53",
    "timezone": "UTC",
    "type": "single",
    "week": "first",
  },
]
`;

exports[`Exception Selectors getStartDate 1`] = `"2018-09-21"`;

exports[`Exception Selectors getStartDateInput 1`] = `"September 21, 2018"`;

exports[`Exception Selectors getStartDateObj 1`] = `2018-09-21T00:00:00.000Z`;

exports[`Exception Selectors getStartTime 1`] = `"14:53"`;

exports[`Exception Selectors getStartTimeNoSeconds 1`] = `"14"`;

exports[`Exception Selectors getTimezone 1`] = `"UTC"`;

exports[`Exception Selectors getType 1`] = `"single"`;

exports[`Exception Selectors getTypeOption 1`] = `
Object {
  "label": "Single Exception",
  "value": "single",
}
`;

exports[`Exception Selectors getWeek 1`] = `"first"`;

exports[`Exception Selectors hasExceptions 1`] = `true`;
